pkg/apis/hive/v1/agent/platform.go

```diff
@@ -0,0 +1,68 @@
+package agent
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Platform stores cluster configuration for an agent based install where hosts are booted and registered, ready
+// to be assigned and used in a cluster.
+type Platform struct {
+	// Cluster name
+	Name string `json:"name"`
+	// Installation will start once cluster is in known state and approved
+	Approved                 bool   `json:"approved,omitempty"`
+	OpenshiftVersion         string `json:"openshiftVersion"`
+	BaseDnsDomain            string `json:"baseDnsDomain,omitempty"`
```

Comment:
Looks like [ClusterDeployment.Spec.BaseDomain](https://github.com/openshift/hive/blob/master/pkg/apis/hive/v1/clusterdeployment_types.go#L75) and could be dropped.